feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/ui/CameraControlsOverlay.kt

```diff
@@ -0,0 +1,429 @@
+) {
+    Column(modifier) {
+        if (showZoomLevel) {
+            ZoomScaleText(zoomLevel)
+        }
+
+        Row(
+            Modifier.fillMaxWidth(),
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            Row(Modifier.weight(1f), horizontalArrangement = Arrangement.SpaceEvenly) {
+                if (!isQuickSettingsOpen) {
+                    FlipCameraButton(
+                        onClick = onFlipCamera,
+                        // enable only when phone has front and rear camera
+                        enabledCondition = currentCameraSettings.isBackCameraAvailable &&
+                                currentCameraSettings.isFrontCameraAvailable
+                    )
+                }
+            }
+            CaptureButton(
+                previewMode = previewMode,
+                isQuickSettingsOpen = isQuickSettingsOpen,
+                videoRecordingState = videoRecordingState,
+                onCaptureImage = onCaptureImage,
+                onCaptureImageWithUri = onCaptureImageWithUri,
+                onToggleQuickSettings = onToggleQuickSettings,
+                onStartVideoRecording = onStartVideoRecording,
+                onStopVideoRecording = onStopVideoRecording
+            )
+            Row(Modifier.weight(1f)) {
+                /*TODO("Place other components here") */
+            }
+        }
+    }
+}
+
+@Composable
+private fun CaptureButton(
+    previewMode: PreviewMode,
+    isQuickSettingsOpen: Boolean,
+    videoRecordingState: VideoRecordingState,
+    modifier: Modifier = Modifier,
+    onCaptureImage: () -> Unit = {},
+    onCaptureImageWithUri: (ContentResolver, Uri?, (PreviewViewModel.ImageCaptureEvent) -> Unit) -> Unit = { _, _, _ -> },
+    onToggleQuickSettings: () -> Unit = {},
+    onStartVideoRecording: () -> Unit = {},
+    onStopVideoRecording: () -> Unit = {}
+) {
+    val multipleEventsCutter = remember { MultipleEventsCutter() }
+    val context = LocalContext.current
+    CaptureButton(
+        modifier = modifier.testTag(CAPTURE_BUTTON),
+        onClick = {
+            multipleEventsCutter.processEvent {
+                when (previewMode) {
+                    is PreviewMode.StandardMode -> {
+                        onCaptureImage()
+                    }
+
+                    is PreviewMode.ExternalImageCaptureMode -> {
+                        onCaptureImageWithUri(
+                            context.contentResolver,
+                            previewMode.imageCaptureUri,
+                            previewMode.onImageCapture
+                        )
+                    }
+                }
+            }
+            if (isQuickSettingsOpen) {
+                onToggleQuickSettings()
+            }
+        },
+        onLongPress = {
+            onStartVideoRecording()
+            if (isQuickSettingsOpen) {
+                onToggleQuickSettings()
+            }
+        },
+        onRelease = { onStopVideoRecording() },
```

Comment:
Somehow this behavior isn't triggering anymore. When you release the record button, it stays active until it is tapped again. I'm not sure why, however.

Reply:
Fixed. It was caused by calling the CaptureButton in two different locations, thereby cancelling the pointer input gesture handling.